### DISCUSSION TITLE
[12 settembre 2024] Esercizio 2

### DISCUSSION THREAD

**Discussion by Nw4pS** [sapienzastudentsnetwork/psd2223#273] [Esami]
Si consideri il seguente circuito combinatorio:

<div align="center">

<img src="https://github.com/user-attachments/assets/2a6a5b22-51c0-405b-b7ef-cdb636342b13" width="60%" height="60%">

</div>

Si scriva l’espressione booleana di f e si scriva f in forma canonica SOP e in forma canonica POS.

<details closed>
    <summary> soluzione A-L </summary>

<div align="center">

![image](https://github.com/user-attachments/assets/10d6b43e-4422-479f-aab5-fb8fde9a9f4d)
![image](https://github.com/user-attachments/assets/dfb746f5-914c-4296-ab35-67fceb7e82d7)
</details>
</div>

**Comment by FeddyLix17** [comment]
considerando le entrate del mux dall'alto verso il basso rispettivamente come 00, 01, 10, 11

$$
    \Large f = \overline{x_3}\overline{x_2}(x_1+x_0) + x_3\overline{x_2}x_1\overline{x_0} + x_3x_2\overline{x_0}
$$


<div align="center">

<table>
<tr>
<td>
<table>
  <tr>
    <th>$\Large x_3$</th>
    <th>$\Large x_2$</th>
    <th>$\Large x_1$</th>
    <th>$\Large x_0$</th>
    <th>$\Large f$</th>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">1</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">1</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">1</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">0</td>
  </tr>
</table>
</td>
<td>
<table>
  <tr>
    <th>$\Large x_3$</th>
    <th>$\Large x_2$</th>
    <th>$\Large x_1$</th>
    <th>$\Large x_0$</th>
    <th>$\Large f$</th>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">1</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">0</td>
    <td align="center">1</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">1</td>
    <td align="center">0</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">0</td>
    <td align="center">1</td>
  </tr>
  <tr>
    <td colspan="5"></td>
  </tr>
    <tr>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">1</td>
    <td align="center">0</td>
  </tr>
</table>
</td>
</tr>
</table>
</div>

forma canonica SOP

$$
    \Large f = \overline{x_3}\overline{x_2}\overline{x_1}x_0 + \overline{x_3}\overline{x_2}x_1\overline{x_0} + \overline{x_3}\overline{x_2}x_1x_0 + x_3\overline{x_2}x_1\overline{x_0} + x_3x_2\overline{x_1}\overline{x_0} + x_3x_2x_1\overline{x_0}
$$

---

forma canonica POS

$$
    \Large 
    \begin{align} 
        f = (x_3 + x_2 + x_1 + x_0) (x_3 + \overline{x_2} + x_1 + x_0) (x_3 + \overline{x_2} + x_1 + \overline{x_0}) (x_3 + \overline{x_2} + \overline{x_1} + x_0) \\ 
        \quad \ \ \ (x_3 + \overline{x_2} + \overline{x_1} + \overline{x_0}) (\overline{x_3} + x_2 + x_1 + x_0) (\overline{x_3} + x_2 + x_1 + \overline{x_0}) (\overline{x_3} + x_2 + \overline{x_1} + \overline{x_0}) \\ 
        \quad \ \ \ (\overline{x_3} + \overline{x_2} + x_1 + \overline{x_0}) (\overline{x_3} + \overline{x_2} + \overline{x_1} + \overline{x_0})
\end{align}
$$
